srtcore/queue.cpp

```diff
@@ -319,6 +293,8 @@ void CSndUList::update(const CUDT* u, EReschedule reschedule)
       }
 
       remove_(u);
+      insert_norealloc_(1, u);
```

Comment:
This seems to be changing behavior of the function. Is this intentional? Is this a bug fix? I don't see any mentioning of that in the PR description. Please verify.

Reply:
This is a bugfix. Actually, this very function call is the same as before, except it was done later, out of the braces.
At this point we can be sure there is a place in the buffer to insert the node to. Because we have removed this socket first. Thus insert function without realloc check is called (`insert_norealloc_`).
But out of this scope block, a regular `insert_` is called, that will check if additional space allocation is required, because we don't know if we have enough space or not.
I was thinking whether it is worth calling simply `insert_` in both cases. That would simplify the code, but add aditional condition check, so might have been a minor burdain on performance if this `update` function is intencively used.

- [ ] Check if `insert_norealloc_` and `remove_` better be combined into one `update_` function.

Reply:
One `update_` function that combines `insert_norealloc_` and `remove_` is actually misleading, because the normal way to update an element in a heap is to remove it and place back again. So my idea was not good.